Add details for kubectl.kubernetes.io/restartedAt annotation (#41348)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

Part of the specification used to implement [ApplySet-based pruning in kubectl](/docs/tasks/manage-kubernetes-objects/declarative-config/#alternative-kubectl-apply-f-directory-prune). This label is what makes an object an ApplySet parent object. Its value is the unique ID of the ApplySet, which is derived from the identity of the parent object itself. This ID **must** be the base64 encoding (using the URL safe encoding of RFC4648) of the hash of the group-kind-name-namespace of the object it is on, in the form: `<base64(sha256(<name>.<namespace>.<kind>.<group>))>`. There is no relation between the value of this label and object UIDs.

### kubectl.kubernetes.io/restartedAt

> Example: `kubectl.kubernetes.io/restartedAt: '2023-03-05T15:30:00Z`

Reviewer: ```suggestion
Example: `kubectl.kubernetes.io/restartedAt: "2023-03-05T15:30:00Z"`
```

Used on: Pod

> This annotation indicates the time when a Kubernetes resource was last restarted using the kubectl command-line tool. It records the time of the most recent workload restart as a Unix timestamp.

Reviewer: ```suggestion
This annotation indicates the time when a workload restart was triggered, using 
the kubectl command-line tool.
The kubelet tool updates the Pod template in a Deployment, ReplicaSet, StatefulSet
or DaemonSet, making a change to the metadata of Pods defined by that template.
By either setting this annotation or changing the value to a new date, the kubectl tool
triggers a rollout and indirectly causes the launch of fresh, replacement Pods.
```

> In the above example, the annotation indicates that the Pod was recently restarted at 3:30 PM UTC on March 31, 2023.

Reviewer: ```suggestion
In the above example, the annotation indicates that the Pod is part of a workload that was
restarted at 3:30 PM UTC on March 31, 2023.
```

> The `"kubectl.kubernetes.io/restartedAt"` annotation is used by Kubernetes to keep track of the timestamp of the most recent container restart for a particular pod. This information can be useful for troubleshooting and auditing purposes, allowing administrators to track when a resource was last restarted and correlate it with other events or changes in the cluster.

Reviewer: Please wrap this line at, say, 100 characters. That helps localization teams.

Reviewer: This doesn't yet look wrapped to me.

Reviewer 2: Any update?

Part of the specification used to implement [ApplySet-based pruning in kubectl](/docs/tasks/manage-kubernetes-objects/declarative-config/#alternative-kubectl-apply-f-directory-prune). This label is what makes an object an ApplySet parent object. Its value is the unique ID of the ApplySet, which is derived from the identity of the parent object itself. This ID **must** be the base64 encoding (using the URL safe encoding of RFC4648) of the hash of the group-kind-name-namespace of the object it is on, in the form: `<base64(sha256(<name>.<namespace>.<kind>.<group>))>`. There is no relation between the value of this label and object UIDs.

### kubectl.kubernetes.io/restartedAt

Example: `kubectl.kubernetes.io/restartedAt: "2023-03-05T15:30:00Z"`

Used on: Pod

This annotation indicates the time when a workload restart was triggered, using 
the kubectl command-line tool.

> The kubelet tool updates the Pod template in a Deployment, ReplicaSet, StatefulSet
> or DaemonSet, making a change to the metadata of Pods defined by that template.

Reviewer 3: ```suggestion
The kubectl tool updates the Pod template in a Deployment, ReplicaSet, StatefulSet
or DaemonSet, making a change to the metadata of Pods defined by that template.
```

By either setting this annotation or changing the value to a new date, the kubectl tool
triggers a rollout and indirectly causes the launch of fresh, replacement Pods.

In the above example, the annotation indicates that the Pod is part of a workload that was

> most recently restarted at 3:30 PM UTC on March 31, 2023.

Reviewer 4: ```suggestion
most recently restarted at 3:30 PM UTC on March 31, 2023.
```
looks like the time stamp is for 3:30pm UTC on March 5, 2023

Author: @reylejano please suggest me what i put here!

Reviewer: Can you work out what the date string `2023-03-05T15:30:00Z` represents? Try https://dencode.com/en/date for example.

Reviewer 5: > Can you work out what the date string `2023-03-05T15:30:00Z` represents? Try https://dencode.com/en/date for example.

Please address thecomment here and make the change

By either setting this annotation or changing the value to a new date, the kubectl tool
triggers a rollout and indirectly causes the launch of fresh, replacement Pods.

> In the above example, the annotation indicates that the Pod was recently restarted at 3:30 PM UTC on March 31, 2023. The `"kubectl.kubernetes.io/restartedAt"` annotation is used by Kubernetes to keep track of the timestamp of the most recent container restart for a particular pod. This information can be useful for troubleshooting and auditing purposes, allowing administrators to track when a resource was last restarted and correlate it with other events or changes in the cluster.

Reviewer 6: please manually wrap long lines like this.

Author: @tengqm : already did this! please check this out.

Reviewer: Hi. This line - 142 at time of writing - is one long line. Put some line breaks in, such as:
```suggestion
In the above example, the annotation indicates that the Pod was recently restarted at
3:30 PM UTC on March 31, 2023. The `"kubectl.kubernetes.io/restartedAt"` annotation is
used by Kubernetes to keep track of the timestamp of the most recent container restart
for a particular pod.
This information can be useful for troubleshooting and auditing purposes, allowing
administrators to track when a resource was last restarted and correlate it with other events
or changes in the cluster.
```

Reviewer: Since the advice isn't accurate, this would be even better:
```suggestion
In the above example, the annotation indicates that the Pod is part of a workload that was
most recently restarted at 3:30 PM UTC on March 31, 2023.

This information can be useful for troubleshooting and auditing purposes, but is mainly used
to trigger the rollout. Recording that date is a side effect of the mechanism to trigger those
changes.
```
